doc/api/process.md

```diff
@@ -1048,6 +1048,11 @@ to the [`Worker`][] constructor. Changes to `process.env` will not be visible
 across [`Worker`][] threads, and only the main thread can make changes that
 are visible to the operating system or to native add-ons.
 
+What's more, use `Object.defineProperty` will only set `value` on target,
+and throw `Error` when define property with getter/setter.
+Because set a accessors or non-configurable/enumerable/writable property is
+not meaningful on `process.env`.
```

Comment:
To be honest, I would not document this, besides adding a short entry to the `changes:` section above (which we should do). In my opinion, it’s not usually helpful to document behaviour for an API when the interaction is not meaningful anyway.